src/averageAges.js

```diff
@@ -20,6 +20,38 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+
+  // ----------------- version 1
+  /*
+  const personFromNeededCentury = century
+    ? people.filter(person => Math.ceil(people.died / 100) === century)
+    : people;
+
+  const maleOnly=personFromNeededCentury.filter(person=>person.sex === 'm');
+
+  const totalAge = maleOnly.reduce(
+    (prev, person) => prev + (person.died - person.born), 0
+  );
+
+  const averageAge = totalAge / maleOnly.length;
+
+  return averageAge;
+  */
+
+  // ----------------- version 2
+
+  const men = people.filter(person => person.sex === 'm')
+    .filter(person => getCentury(person) === century || !century);
+
+  const totalAge = men.reduce(
+    (sum, person) => sum + (person.died - person.born), 0
+  );
+
+  return totalAge / men.length;
+}
+
+function getCentury(person) {
+  return Math.ceil(person.died / 100);
```

Comment:
Consider using constants instead of magic numbers

Reply:
fixed